src/Logger.ts

```diff
@@ -0,0 +1,100 @@
+let winston: any = require('winston');
+
+/**
+ * Represents log level.
+  <ul>
+  <li>LogLevel.Error: error level.</li>
+  <li>LogLevel.Warn: warn level. </li>
+  <li>LogLevel.Info: info level.</li>
+  <li>LogLevel.Verbose: verbose level.</li>
+  <li>LogLevel.Debug: debug level.</li>
+  <li>LogLevel.Silly: silly level.</li>
+  </ul>
+ */
+export const LogLevel = {
+    Error: "error",
+    Warn: "warn",
+    Info: "info",
+    Verbose: "verbose",
+    Debug: "debug",
+    Silly: "silly"
+}
+
+/**
+ * Represents Logger for SDK.
+ */
+export class Logger {
+    private static instance: Logger;
+    private winstonLogger: any;
+
+    private constructor() {
+        this.winstonLogger = new winston.Logger({
+            level: "error",
+            transports: [
+                new (winston.transports.Console)()
+            ]
+        })
+    }
+
+    /**
+     * Get a singleton Logger instance.
+     * @return {Logger} returns singleton instance of Logger.
+     * @example
+     * // should set log level first
+     * ThingIF.Logger.getInstance().setLogLevel(ThingIF.LogLevel.Debug);
+     */
+    static getInstance(): Logger {
+        if(!Logger.instance) {
+            Logger.instance = new Logger();
+        }
+        return Logger.instance;
+    }
+
+    /**
+     * Set level of logger.
+     * @param {string} level level should be one of values of const {@link LogLevel}.
+     */
+    setLogLevel(level: string) {
+        this.winstonLogger.level = level;
+    }
+
+    /**
+     * Make a log record.
+     * @param {string} level level should be one of values of const {@link LogLevel}.
+     * @param {string} msg log message.
+     */
+    log(level: string, msg: string) {
```

Comment:
can you support [string-interpolation](https://github.com/winstonjs/winston#string-interpolation)?